ark/type/methods/base.ts

```diff
@@ -140,11 +141,17 @@ interface Type<out t = unknown, $ = {}>
 	optional<r = applyConstraint<t, Optional>>(): instantiateType<r, $>
 
 	default<
-		const value extends this["inferIn"],
+		const value extends Extract<this["inferIn"], Primitive>,
 		r = applyConstraint<t, Default<value>>
 	>(
 		value: value
-	): instantiateType<r, $>
+	): NoInfer<instantiateType<r, $>>
+	default<
+		const value extends this["inferIn"],
+		r = applyConstraint<t, Default<value>>
+	>(
+		value: () => value
+	): NoInfer<instantiateType<r, $>>
```

Comment:
should just use `value: DefaultFor<value>` here, having overload with `never` breaks autocompletion

Reply:
Seems good to use the same type util everywhere anyways